android/src/com/mapswithme/maps/editor/EditorFragment.java

```diff
@@ -276,6 +328,34 @@ private boolean validateFields()
       return false;
     }
 
+    if (!Editor.nativeIsFacebookPageValid(mFacebookPage.getText().toString()))
+    {
+      mFacebookPage.requestFocus();
+      InputUtils.showKeyboard(mFacebookPage);
+      return false;
+    }
+
+    if (!Editor.nativeIsInstagramPageValid(mInstagramPage.getText().toString()))
+    {
+      mInstagramPage.requestFocus();
+      InputUtils.showKeyboard(mInstagramPage);
+      return false;
+    }
+
+    if (!Editor.nativeIsTwitterPageValid(mTwitterPage.getText().toString()))
+    {
+      mTwitterPage.requestFocus();
+      InputUtils.showKeyboard(mTwitterPage);
+      return false;
+    }
+
+    if (!Editor.nativeIsVkPageValid(mVkPage.getText().toString()))
+    {
+      mVkPage.requestFocus();
+      InputUtils.showKeyboard(mVkPage);
+      return false;
+    }
```

Comment:
NIT: All this routing can be prettified if we declare arrays instead of separate vars:

enum PageID { ... }
EditText mPages[4]
TextInputLayout mInputPages[4]
nativeIsValidPage(pageID)
nativeSetPageValue(pageID)

Of course not in this patch. Can be refactored later and _generalazied_ for all optional string values.

Reply:
@vng Yes, I did a lot of copy-n-paste code for each social network. Some refactoring would be great to have :)

Reply:
@strump Thanks a lot. Your patch is good!